File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int f = 0;
        int l = nums.length-1;
        while(f<=l) {
            int mid = (f+l)/2;
            if(nums[mid] == target) {
                return mid;
            } else if(target > nums[mid]) {
                f = mid+1;
            } else if(target < nums[mid]) {
                l = mid -1;
            }
        }
        return -1;
    }
}

File: 0792-binary-search/0792-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let f=0;
    let l=nums.length-1;
    while(f<=l) {
        let mid = Math.floor((f+l)/2);
        if(nums[mid] === target) return mid;
        else if(nums[mid] > target) l=mid-1;
        else if(nums[mid] < target) f=mid+1;
    }
    return -1;
};